back-end/services/registerService.js

```diff
@@ -0,0 +1,16 @@
+const userModel = require('../models/userModel');
+
+const createUser = async (userInfo) => {
+  const { name, email, password, role } = userInfo;
+
+  const result = await userModel.getByEmail(email);
+  if (result) {
+    return { error: true, message: 'Email already exists ' };
```

Comment:
```suggestion
    return { error: 'email_in_use' };
```
E ai lá no errorHandler, você pode criar um atributo assim :
email_in_use: { message: 'Email already exists ', status: 409 }

Assim o erro quando for pro next('email_in_use'), o handler vai criar um erro com a messagem e o status, e envia-lo.
O que acha?

Reply:
Pois é... não mexi mais depois daquela sugestão sua. Irei corrigir junto com essas issues. Vlw!